src/web/core/src/main/java/org/geoserver/web/wicket/HTTPURLsListTextArea.java

```diff
@@ -0,0 +1,101 @@
+
+/** TextArea for specifying a list of HTTP Urls (separated by new lines). */
+public class HTTPURLsListTextArea extends TextArea<List<String>> {
+
+    private static final long serialVersionUID = -8195179437229644665L;
+
+    public HTTPURLsListTextArea(String id, IModel<List<String>> model) {
+        super(id, model);
+
+        add(new HTTPURLsListValidator());
+        setType(List.class);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <C> IConverter<C> getConverter(Class<C> type) {
+        if (List.class.isAssignableFrom(type)) {
+            return (IConverter<C>) new URLsListConverter();
+        }
+        return super.getConverter(type);
+    }
+
+    private static class URLsListConverter implements IConverter<List<String>> {
+        private static final long serialVersionUID = 1083795866666107798L;
+
+        static final Pattern NEW_LINE_SEPARATED =
+                Pattern.compile("\\s*\\r?\\n\\s*", Pattern.MULTILINE);
+
+        @Override
+        public String convertToString(List<String> urlsList, Locale locale) {
+            if (urlsList.isEmpty()) return "";
+
+            StringBuffer sb = new StringBuffer();
+            for (String url : urlsList) {
+                sb.append(url).append("\n");
+            }
+            sb.setLength(sb.length() - 1);
+            return sb.toString();
+        }
+
+        @Override
+        public List<String> convertToObject(String value, Locale locale) {
+            if (value == null || value.trim().equals("")) return Collections.emptyList();
+            return new ArrayList<>(Arrays.asList(NEW_LINE_SEPARATED.split(value)));
+        }
+    }
+
+    private static class HTTPURLsListValidator implements IValidator<List<String>> {
+
+        private static final long serialVersionUID = 8041469734553805086L;
+
+        @Override
+        public void validate(IValidatable<List<String>> validatable) {
+            List<String> urlsList = validatable.getValue();
+            List<String> invalid = new ArrayList<>();
+            urlsList.stream()
+                    .forEach(
+                            (url) -> {
+                                boolean isValid = false;
+                                try {
+                                    // Only valid URLs and HTTP Urls are supported
+                                    URL validURL = new URL(url);
+                                    isValid = validURL.getProtocol().startsWith("http");
+                                } catch (Exception e) {
+                                    isValid = false;
+                                }
+                                if (!isValid) {
+                                    invalid.add(url);
+                                }
+                            });
+
+            if (!invalid.isEmpty()) {
+                IValidationError err =
+                        new ValidationError("HTTPURLsListTextArea.invalidURL")
```

Comment:
This constructor takes a message rather than a i18n key... wondering if you could not just get away with the empty argument constructor here?

Reply:
Sorry. I was "copying" from SRSListArea which was using the i18n key.
I will set it empty.